[docs] EAS Build: Improve creating your first build docs (#18092)
expo/expo: docs/pages/build/setup.md

For a small app, builds for Android and iOS platforms trigger within a few minutes. If you encounter any issues along the way, you can reach out on the [Expo forums](https://forums.expo.dev/) or [Discord](https://chat.expo.dev/).

## Prerequisites

EAS Build is a new and rapidly evolving service. Before you set out to create a build for your project we recommend consulting the [limitations](/build-reference/limitations) page and the other prerequisites below.

<Collapsible summary="A React Native iOS and/or Android project that you want to build">

Don't have a project yet? No problem. It's quick and easy to create a "Hello world" app that you can use with this guide.

- Install Expo CLI by running the command:

<Terminal cmd={['$ npm install -g expo-cli']} />

- Run the following command to create a new project:

<Terminal cmd={['$ npx create-expo-app my-app']} />

EAS Build also works well with projects created by `npx create-react-native-app`, `npx react-native`, `ignite-cli`, and other project bootstrapping tools.

</Collapsible>

<Collapsible summary="An Expo user account">

> EAS Build is available to everybody with an Expo account. You can sign up at [https://expo.dev/signup](https://expo.dev/signup). It is available for both free and paid subscriptions. 

Reviewer: ```suggestion
EAS Build is available to anyone with an Expo account, regardless of whether you pay for EAS or use our free tier. You can sign up at [https://expo.dev/signup](https://expo.dev/signup).
```

Paid subscribers get quality improvements such as additional build concurrencies, priority access to minimize the time your builds spend queueing, and increased limits on build timeouts. Learn more about different plans and benefits at [EAS pricing](https://expo.dev/pricing).

</Collapsible>

## 1. Install the latest EAS CLI

EAS CLI is the command-line app that you will use to interact with EAS services from your terminal. To install it, run the command:

> <Terminal cmd={['$ npm install -g expo-cli']} />

Reviewer: ```suggestion
<Terminal cmd={['$ npm install -g eas-cli']} />
```

> You can also use the above command to check if a new version of the EAS CLI app is available. We encourage you to always stay up to date with the latest version.

Reviewer: ```suggestion
You can also use the above command to check if a new version of EAS CLI is available. We encourage you to always stay up to date with the latest version.
```

> We recommend using `npm` instead of `yarn` for global package installations. You may alternatively use `npx eas-cli`, just remember to use that instead of `eas` whenever it's called for in the documentation.

## 2. Log in to your Expo account

> If you are already signed in to Expo account using Expo CLI, you can skip the steps described in this section. If you are not, run the following command to log in:

Reviewer: ```suggestion
If you are already signed in to an Expo account using Expo CLI, you can skip the steps described in this section. If you are not, run the following command to log in:
```

- Does your app code depend on environment variables? [Add them to your build configuration](/build-reference/variables).
- Is your project inside of a monorepo? [Follow these instructions](/build-reference/how-tos#how-to-set-up-eas-build-with).
- Do you use private npm packages? [Add your npm token](/build-reference/private-npm-packages).
- Does your app depend on specific versions of tools like Node, Yarn, npm, CocoaPods, or Xcode? [Specify these versions in your build configuration](/build/eas-json).

## 4. Run a build

### Build for Android device/emulator or iOS simulator

The easiest way to try out EAS Build is to create a build that you can run on your Android device/emulator or iOS simulator. It's quicker than uploading it to a store, and you don't need store developer membership accounts. If you'd like to try this, read about [creating an installable APK for Android](/build-reference/apk) and [creating a simulator build for iOS](/build-reference/simulators).

### Build for app stores

Before the build process can start for app stores, you will need to have a store developer account and generate or provide app signing credentials.

> Whether you have experience with generating app signing credentials or not, EAS CLI does the heavy lifting. You can opt-in for EAS CLI to handle the generating app signing credentials process. Check out the steps for [Android app signing credentials](#android-app-signing-credentials) or [iOS app signing credentials](#ios-app-signing-credentials) process below for more information.

Reviewer: ```suggestion
Whether you have experience with generating app signing credentials or not, EAS CLI does the heavy lifting. You can opt-in for EAS CLI to handle the app signing credentials process. Check out the steps for [Android app signing credentials](#android-app-signing-credentials) or [iOS app signing credentials](#ios-app-signing-credentials) process below for more information.
```
